### PR TITLE
Shipwire Grabber: Improve tooling

### PR DESCRIPTION
**Pull request by andyg0808**
This sets up [Pipenv](https://pipenv.pypa.io/en/latest/) as the primary dependency manager and configures some development dependencies, in order to standardize some tooling around this Python code.

The Dockerfile wouldn't build on the latest Alpine; it appears `pip` doesn't get completely set up during the installation of the `python3` package. Now we set it up as part of the Dockerfile.

This pull also adds `build.sh` and `test.sh` scripts in order to document how to run the new tooling. The `build.sh` script in particular is important, as it ties together the dependency list in the Pipfile and the list in the `requirements.txt` that is installed in the Docker container.

### CODE REVIEW

**Comment by andyg0808** [comment]
The `build.sh` and `test.sh` scripts run without errors. I'm going to count that as QA :+1: for now.

File: Dockerfile [hunk 1]
```diff
@@ -7,6 +7,8 @@ COPY . /opt/shipwire-analytics-grabber
 WORKDIR  /opt/shipwire-analytics-grabber
 
 # Install python project, install modules
-RUN      pip3 install -r requirements.txt
+RUN      python3 -m ensurepip \
+         && pip3 install --upgrade pip \
+         && pip3 install -r requirements.txt
```
Comment (djmetzle) [inline Dockerfile]: Weird align on the continuations here.

Do you mind adding in a space to be a little more consistent with the shell style in our other repos?
```suggestion
          && pip3 install --upgrade pip \
          && pip3 install -r requirements.txt
```

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
+[[source]]
+name = "pypi"
```
Comment (djmetzle) [inline Pipfile:2]: Ehw! Pipfile! :sweat_smile:

File: build.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+pipenv lock -r --pre > requirements.txt
```
Comment (djmetzle) [inline build.sh]: Is this going to generate untracked changes when it runs?

File: test.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+pip install pipenv
+pipenv install
+pipenv run pytest
```
Comment (djmetzle) [inline test.sh:6]: While we're here, do you want to run `black`? :smiley:

**Comment by djmetzle** [comment]
CR :fist_raised:

**Review by djmetzle — APPROVED** [review]

File: test.sh (hunk 4 above)
Comment (andyg0808) [reply test.sh:6]: Yeah, although maybe in a way that doesn't make uncommitted changes by accident.

File: build.sh (hunk 3 above)
Comment (andyg0808) [reply build.sh]: Swapped `requirements.txt` for `pipenv` entirely, per our conversation.
f4cf624

File: test.sh (hunk 4 above)
Comment (andyg0808) [reply test.sh:6]: 022ae61

**Comment by djmetzle** [comment]
Nice! CR :gear:

**Comment by andyg0808** [comment]
QA :+1: as before: I reran the test and build scripts, and they're not failing.

**Comment by sctice** [comment]
CR 🎏

**Comment by sctice** [comment]
Although it makes complete sense here, in the Pulldasher context, "Improve tooling" is a pretty generic name. I think it would help to provide a prefix like "Shipwire Import: improve tooling".

**Comment by djmetzle** [comment]
Meh, "improve tooling" makes sense in the context of the one PR on this repo. Its only problematic in the pulldasher grab-bag. :stuck_out_tongue_closed_eyes:

**Pull request merged**